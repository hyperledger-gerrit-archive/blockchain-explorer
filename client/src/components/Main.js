import React from 'react';
import { HashRouter as Router, Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import createStore from '../store/index';
import Layout from './Layout/index';

const Main = () =>
  (
      <Router>
        <div className="App">
          <Switch>
            <Route exact path='/' component={Layout} />
          </Switch>
        </div>
      </Router>
  )

export default Main
