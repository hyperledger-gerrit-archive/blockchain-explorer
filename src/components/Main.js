import React from 'react';
import { HashRouter as Router, Route, Switch } from 'react-router-dom'
import { Provider } from 'react-redux'
import createStore from '../store'
import Channel from './Channel/Channel'

const Main = () =>
(<Provider store={createStore()} >
      <Router>
        <div className="App">
          <Switch>
          <Route exact path='/' component={Channel}/>
          </Switch>
        </div>
      </Router>
    </Provider>)

export default Main
