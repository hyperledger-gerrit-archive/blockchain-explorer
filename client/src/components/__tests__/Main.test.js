import React from "react";
import {
    mount,
    shallow
} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import Main from "../Main";
import renderer from 'react-test-renderer';

const middlewares = [];
const mockStore = configureStore(middlewares);
const initialState = {};

describe('Testing Main component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow( <Main /> , {
            context: {
                store: mockStore(initialState)
            }
        }, );
        expect(wrapper).toMatchSnapshot();
    })
    it('wrapper length > 0 ', () => {
        expect(wrapper.length).toEqual(1);
    });
});