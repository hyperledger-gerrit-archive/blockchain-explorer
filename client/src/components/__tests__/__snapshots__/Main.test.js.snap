// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Main component wrapper length > 0  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <HashRouter>
        <div
                className="App"
        >
                <Switch>
                        <Route
                                component={[Function]}
                                exact={true}
                                path="/"
                        />
                </Switch>
        </div>
</HashRouter>,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="App"
>
          <Switch>
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/"
                    />
          </Switch>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Switch>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
</Switch>,
          "className": "App",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <HashRouter>
          <div
                    className="App"
          >
                    <Switch>
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/"
                              />
                    </Switch>
          </div>
</HashRouter>,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="App"
>
            <Switch>
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/"
                        />
            </Switch>
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Switch>
              <Route
                            component={[Function]}
                            exact={true}
                            path="/"
              />
</Switch>,
            "className": "App",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
